### PR TITLE
finalize Drain Mana spells

### PR DESCRIPTION
**Pull request by Tobschinski**

### CODE REVIEW

**Comment by killerwife** [comment]
Nice to see that my change was of use.

**Comment by cala** [comment]
> Nice to see that my change was of use.
You mean avoid using `TRIGGERED_OLD_TRIGGERED`?

Aside from this, the change looks fine to me and I will merge it if none objects.

**Comment by killerwife** [comment]
No, I moved handling of SpellInfo MaxAffectedTargets AFTER CheckTarget, which enabled us to have a spell which has target count restricted at the same time as it being restricted to mana users.

**Comment by cala** [comment]
Ok, then I missed the whole point of your comment. 👅 
Let's kick me!

**Comment by Tobschinski** [comment]
TRIGGERED_OLD_TRIGGERED is obsolete?

**Comment by killerwife** [comment]
Nope, I just gave it a really bad name :D

**Comment by cala** [comment]
Let's kick you too.

**Comment by Tobschinski** [comment]
Well, there will be someone in about 10 years who will take care of all cases.

File: src/game/Spells/SpellAuras.cpp [hunk 1]
```diff
@@ -1285,6 +1285,16 @@ void Aura::HandleAuraDummy(bool apply, bool Real)
                 if (Unit* caster = GetCaster())
                     caster->ModifyHealth(caster->GetMaxHealth() - caster->GetHealth());
                 return;
+            case 25185:                                     // Itch
```
Comment (Tobschinski) [inline src/game/Spells/SpellAuras.cpp]: This spell has a scripted Effect self-cast, no clue what it does. Maybe it is related to the Catalyst cast, maybe it isn't.

File: src/game/Spells/SpellAuras.cpp [hunk 2]
```diff
@@ -1285,6 +1285,16 @@ void Aura::HandleAuraDummy(bool apply, bool Real)
...
+            {
+                GetCaster()->CastSpell(target, 25187, TRIGGERED_OLD_TRIGGERED);
+                return;
+            }
+            case 26077:                                     // Itch
```
Comment (Tobschinski) [inline src/game/Spells/SpellAuras.cpp]: same

**Comment by cala** [comment]
@Tobschinski what is your opinion on this PR. Do you still need to work a bit more on it or can we review and merge it?

**Comment by Tobschinski** [comment]
I don't believe in work.

**Comment by cala** [comment]
> I don't believe in work.

I take that as a "You can merge it". Feel free to throw a trout at me if I misunderstood you. Fishes are running low on my plate of late.

**Comment by cala** [comment]
Unless new feedback or suggestion is made, I will merge this next week.

**Comment by cala** [comment]
Merged. Thank you!

**Pull request closed without merging**